app/records/related.py

```diff
@@ -0,0 +1,203 @@
+import logging
+import random
+from typing import List
+
+from app.records.constants import TNA_LEVELS
+from app.records.models import Record
+from app.search.api import search_records
+
+logger = logging.getLogger(__name__)
+
+_LEVEL_FILTERS_SERIES_TO_ITEM = [
+    f"level:{TNA_LEVELS[str(i)]}" for i in range(3, 8)
+]
+
+
+def get_related_records_by_subjects(
+    current_record: Record, limit: int = 3
+) -> list[Record]:
+    """
+    Fetches related records that share subjects with the current record.
+    Returns a random selection from matching records.
+    Only applies to TNA records (Non-TNA records don't have subjects).
+
+    Args:
+        current_record: The record to find relations for
+        limit: Maximum number of related records to return (default 3)
+
+    Returns:
+        List of related Record objects (random selection), or empty list if none found
+    """
+    # Only TNA records have subjects
+    if not current_record.is_tna or not current_record.subjects:
+        return []
```

Comment:
The function name `get_related_records_by_subjects` doesn’t indicate that it checks `current_record.is_tna`.
It might be clearer to either include `_tna` in the module and function names, or move the `is_tna` check outside this module.